thread_test.go

```diff
@@ -0,0 +1,43 @@
+package sortthread
+
+import "testing"
+
+var threadTests = []struct {
+	name     string
+	expected string
+	thread   []interface{}
+}{
+	{
+		name:     "simple",
+		expected: "(1 2 3 4)",
```

Comment:
Honestly I'm not a fan of this representation for tests. It's not clear the algorithm is doing the right thing. I'd prefer an explicit Go `Thread` struct instead (and probably `reflect.DeepEqual`).